ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlAbstractDecoder.kt

```diff
@@ -123,6 +143,26 @@ public abstract class TomlAbstractDecoder : AbstractDecoder() {
         }
     }
 
+    private inline fun <reified T> decodeUnsignedPrimitiveType(): T {
+        val keyValue = decodeKeyValue()
+        try {
+            return when (val value = keyValue.value) {
+                is TomlLong -> decodeUnsignedInteger((value.content as Long).toULong(), keyValue.lineNo)
```

Comment:
hm, why for TomlLong we have `decodeUnsignedInteger`? It should be signed, no?

Reply:
```kotlin
 * TomlUnsignedLong - Toml AST Node for a representation of Arbitrary 64-bit unsigned integers: key = 9_223_372_036_854_775_808
 * This node contains integer in range: [2^63, 2^64-1]. Integers less than 2^63 are treated as TomlLong
```
On parsing level we just see the number and don't know it's type